Lib/typing.py

```diff
@@ -433,6 +433,9 @@ def __reduce__(self):
     def __call__(self, *args, **kwds):
         raise TypeError(f"Cannot instantiate {self!r}")
 
+    def __iter__(self):
+        raise TypeError(f"{self} does not support iteration")
```

Comment:
```suggestion
    __iter__ = None
```

E.g.

```python
>>> class Bar:
...     def __getitem__(self, x):
...         return 'foo'
...     __iter__ = None
...
>>> b = Bar()
>>> list(b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'Bar' object is not iterable
>>> class Foo:
...     def __getitem__(self, x):
...         return 'foo'
...
>>> list(Foo())  #  This hangs forever
```

Reply:
Oh woops, I didn't see @JelleZijlstra's review until I finished suggesting this.

Reply:
Thanks for demonstrating that the error message is good!

Reply:
Switched to using `__iter__ = None`.